### ISSUE TITLE
What is padding_causal?

### ISSUE THREAD

**Issue by 1049451037** [NVIDIA/TransformerEngine#908]
What is the meaning of padding_causal? Now that we are using causal, the tokens in the left cannot see the tokens in the right. So the paddings are not necessarily to be distinguished in the attention mask. Why do we need padding_causal?

Because I met the issue in the `transformer_engine/pytorch/attention.py` when I pass in `causal`.

https://github.com/NVIDIA/TransformerEngine/blob/086a12fe9cfade2d49eaa388b991d397e8168477/transformer_engine/pytorch/attention.py#L3720-L3722

**Comment by wplf** [comment]
That confuses me, too.

**Comment by wplf** [comment]
Hello，this may help you.
https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/c/fused_attn.html

**Comment by sbhavani** [comment]
`padding_causal` means that both masks are applied: the causal/autoregressive mask and the padding or sequence-boundary mask.

The current PyTorch docs describes it here:
https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/attention/dot_product_attention/dot_product_attention.py

Closing as answered.
